builtin/providers/azurerm/resource_arm_managed_disk.go

```diff
@@ -0,0 +1,256 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/arm/disk"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+	"net/http"
+	"strings"
+)
+
+func resourceArmManagedDisk() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedDiskCreate,
+		Read:   resourceArmManagedDiskRead,
+		Update: resourceArmManagedDiskCreate,
+		Delete: resourceArmManagedDiskDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"storage_account_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(disk.PremiumLRS),
+					string(disk.StandardLRS),
+				}, true),
+			},
+
+			"create_option": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(disk.Import),
+					string(disk.Empty),
+					string(disk.Copy),
+				}, true),
+			},
+
+			"source_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"source_resource_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"os_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(disk.Windows),
+					string(disk.Linux),
+				}, true),
+			},
+
+			"disk_size_gb": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validateDiskSizeGB,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func validateDiskSizeGB(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(int)
+	if value < 1 || value > 1023 {
```

Comment:
From looking at the [docs](https://docs.microsoft.com/en-us/azure/storage/storage-managed-disks-overview) it seems the valid disk dizes are:
- premium: 128, 512, 1024
- standard: 32, 64, 128, 512, 1024
Shouldn't this be checked here?